IPython/kernel/zmq/tests/test_session.py

```diff
@@ -100,7 +100,7 @@ def test_send(self):
         ident, msg_list = self.session.feed_identities(B.recv_multipart())
         new_msg = self.session.deserialize(msg_list)
         self.assertEqual(ident[0], b'foo')
-        self.assertEqual(new_msg['msg_id'],msg['msg_id'])
+        self.assertEqual(new_msg['msg_id'],header['msg_id'])
         self.assertEqual(new_msg['msg_type'],msg['msg_type'])
```

Comment:
the entire header is already compared below. This is testing that the msg_id is copied from the header to top-level on recv (a minor convenience copy).